src/main/java/org/alfresco/metrics/rest/RestMetricsReporterImpl.java

```diff
@@ -0,0 +1,197 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ * 
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ * #L%
+ */
+package org.alfresco.metrics.rest;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import org.alfresco.micrometer.MetricsController;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.springframework.beans.BeansException;
+import org.springframework.beans.factory.InitializingBean;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.ApplicationContextAware;
+
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Tag;
+
+public class RestMetricsReporterImpl implements RestMetricsReporter, ApplicationContextAware, InitializingBean
+{
+    private Log logger = LogFactory.getLog(getClass());
+    public static final String METER_REGISTRY_BEAN_ERROR_MESSAGE = "Could not get the meterRegistry bean, essential for reporting REST metrics.";
+    private static final String REST_EXECUTION_TIME = "rest-api.execution.time";
+    private static final int MAX_TAG_LENGTH = 1024;
+
+    private boolean restMetricsEnabled;
+    private boolean restServicePathMetricsEnabled;
+
+    private ApplicationContext applicationContext;
+    private MeterRegistry meterRegistry;
+
+    @Override
+    public boolean isRestServicePathMetricsEnabled()
+    {
+        return restServicePathMetricsEnabled;
+    }
+
+    public void setRestMetricsEnabled(boolean restMetricsEnabled)
+    {
+        this.restMetricsEnabled = restMetricsEnabled;
+    }
+
+    public void setRestServicePathMetricsEnabled(boolean restServicePathMetricsEnabled)
+    {
+        this.restServicePathMetricsEnabled = restServicePathMetricsEnabled;
+    }
+
+    @Override
+    public boolean isRestMetricsEnabled()
+    {
+        return restMetricsEnabled;
+    }
+
+    @Override
+    public void afterPropertiesSet() throws Exception
+    {
+        try
+        {
+            init();
+        }
+        catch (Exception e)
+        {
+            if (logger.isWarnEnabled())
+            {
+                logger.warn("Could not initialize rest metrics reporter: " + e.getMessage(), e);
+            }
+        }
+
+    }
+
+    private void init()
+    {
+        if (!isRestMetricsEnabled())
+        {
+
+            if (logger.isDebugEnabled())
+            {
+                logger.debug("Rest Metrics reporting is not enabled");
+            }
+            return;
+        }
+
+        MetricsController metricsController = null;
+
+        try
+        {
+            metricsController = (MetricsController) applicationContext.getBean("metricsController");
```

Comment:
get the been from spring : https://github.com/Alfresco/alfresco-repository/blob/master/src/main/resources/alfresco/core-services-context.xml#L1373